litestar/_openapi/plugin.py

```diff
@@ -43,13 +49,52 @@ def _build_openapi_schema(self) -> OpenAPI:
 
     def provide_openapi(self) -> OpenAPI:
         if not self._openapi_schema:
-            self._openapi_schema = self._build_openapi_schema()
+            self._openapi_schema = self._build_openapi()
         return self._openapi_schema
 
+    @staticmethod
+    def provide_openapi_schema(__openapi: Any) -> Dict[str, Any]:  # noqa: UP006
+        return __openapi.to_schema()  # type: ignore[no-any-return]
+
+    def create_openapi_router(self) -> Router:
+        def create_handler(plugin: OpenAPIRenderPlugin) -> HTTPRouteHandler:
+            paths = [plugin.path] if isinstance(plugin.path, str) else list(plugin.path)
+            if any(path.endswith(self.openapi_config.root_schema_site) for path in paths):
+                paths.append("/")
```

Comment:
This is how I'm handling honoring the `OpenAPIConfig.root_schema_site` config.. a bit ugly - but in the future we could just make this the first registered render plugin.